src/components/MovieInformationContent.ts

```diff
@@ -0,0 +1,106 @@
+    this.init();
+    this.image = $<HTMLImageElement>('.information-image');
+    this.title = $<HTMLHeadingElement>('.information-title');
+    this.metaInfo = $<HTMLParagraphElement>('.information-meta-info');
+    this.voteAverage = $<HTMLParagraphElement>('.information-vote-average-rate');
+    this.overview = $<HTMLParagraphElement>('.information-overview');
+  }
+
+  static getInstance(): MovieInformationContent {
+    if (!MovieInformationContent.instance) {
+      MovieInformationContent.instance = new MovieInformationContent();
+    }
+
+    return MovieInformationContent.instance;
+  }
+
+  private template() {
+    return `
+      <img class="information-image" src="" loading="lazy" alt="" />
+      <div class="information-container">
+        <h3 class="information-title"></h3>
+        <p class="information-meta-info margin-bottom-6"></p>
+        <p class="information-vote-average-rate"></p>
+        <div class="hr"></div>
+        <h6 class="information-sub-title">내 별점</h6>
+        <div class="information-user-vote">
+          <div class="vote-stars"></div>
+          <div class="vote-stars--temp hide"></div>
+          <p class="vote-message"></p>
+          <p class="vote-info"></p>
+        </div>
+        <div class="hr"></div>
+        <h6 class="information-sub-title">줄거리</h6>
+        <p class="information-overview"></p>
+      </div>`;
+  }
+
+  private init() {
+    MovieList.on(MOVIE_RETRIEVED, (event) => {
+      const { movie } = (event as CustomEvent).detail;
+      this.render(movie);
+    });
+  }
+
+  private renderPosterImage(title: string, imagePath: string) {
+    this.image.src = `${POSTER_BASE_URL}${imagePath}`;
+    this.image.alt = title;
+  }
+
+  private renderTitle(title: string) {
+    this.title.textContent = title;
+  }
+
+  private renderMetaInfo(releaseDate: string, genres: string[]) {
+    this.metaInfo.textContent = `${releaseDate.split('-')[0]} · ${genres.join(', ')}`;
+  }
+
+  private renderVoteAverage(voteAverage: number) {
+    const template = `
+      평균 
+      <img class="vote-average-star" src="${voteAverage ? FilledStar : EmptyStar}" alt="별점" />
+      <span class="vote-average">${voteAverage}</span>
+    `;
+
+    this.voteAverage.replaceChildren();
+    this.voteAverage.insertAdjacentHTML('beforeend', template);
+  }
+
+  private renderOverview(overviewText: string) {
+    if (overviewText) {
+      this.overview.textContent = overviewText;
+      this.overview.classList.remove('secondary-text');
+      return;
+    }
+
+    this.overview.textContent = EMPTY_OVERVIEW_MESSAGE;
+    this.overview.classList.add('secondary-text');
+  }
+
+  render(movie: Movie) {
+    this.renderTitle(movie.title);
+    this.renderPosterImage(movie.title, movie.posterPath);
+    this.renderMetaInfo(movie.releaseDate, movie.genres);
+    this.renderVoteAverage(movie.voteAverage);
+    this.renderOverview(movie.overview);
+  }
+}
+
+export default MovieInformationContent.getInstance();
```

Comment:
```suggestion
const instance = new MovieInformationContent()

export default instance;
```

이렇게 하는 방법도 있는데, getInstance 스태틱 메소드를 사용해서 애슐리가 나타내고 싶었던건 무엇인가요? 
(90% 알거 같은데, 그래도 한번 더 애슐리의 말로 들어보고 싶네요)

Reply:
`getInstance` 스태틱 메서드를 사용한 이유는 인스턴스를 진짜 정말 딱 한 번만 만들어서 사용하고 싶어서입니다! 처음에 객체로 컴포넌트를 만들었던 이유 중 하나도 새로운 인스턴스를 만들 걱정 없이 사용할 수 있다는 것이었습니다. 그래서 클래스로 변경하면서도 싱글톤 패턴을 사용해서 인스턴스 하나만 사용하고자 했습니다. 

`const instance = new MovieInformationContent()` 적는 것은 import할 때마다 인스턴스를 생성한다고 이해하고 있는 데 맞나요?

Reply:
> const instance = new MovieInformationContent() 적는 것은 import할 때마다 인스턴스를 생성한다고 이해하고 있는 데 맞나요?

결론부터 말하면 하나의 인스턴스밖에 생성되지 않습니다.

이는 브라우저에서 js를 실행할때 import 구문을 어떻게 실행하는지 연관되는데요~

자세한건 mdn 선생님께 넘길게요 ㅎㅎ

Reply:
디자인 패턴들 유명한거 많은데, JS에서는 그런 복잡한 패턴을 안써도 해결할 수 있는 문제들이 꽤 있어요.

그리고 함수형 프로그래밍을 하면 또 해결되는 문제들도 있구요.

JS를 아주아주아주아주 깊숙하게 이해해야 합니다. 그래야 오-래, 멀-리 가는 FE개발자가 됩니다~